agent/backend/otel/utils.go

```diff
@@ -0,0 +1,96 @@
+package otel
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/netboxlabs/orb-agent/agent/backend"
+)
+
+func (d *openTelemetryBackend) getProcRunningStatus() (backend.RunningStatus, string, error) {
+	if d.proc == nil {
+		return backend.Unknown, "backend not started yet", nil
+	}
+	status := d.proc.Status()
+
+	if status.Error != nil {
+		errMsg := fmt.Sprintf("device-discovery process error: %v", status.Error)
+		return backend.BackendError, errMsg, status.Error
+	}
+
+	if status.Complete {
+		err := d.proc.Stop()
+		return backend.Offline, "device-discovery process ended", err
+	}
+
+	if status.StopTs > 0 {
+		return backend.Offline, "device-discovery process ended", nil
+	}
+	return backend.Running, "", nil
+}
+
+// note this needs to be stateless because it is called for multiple go routines
+func (d *openTelemetryBackend) request(url string, payload interface{}, method string, body io.Reader, contentType string, timeout int32) error {
+	client := http.Client{
+		Timeout: time.Second * time.Duration(timeout),
+	}
+
+	status, _, err := d.getProcRunningStatus()
+	if status != backend.Running {
+		d.logger.Warn("skipping device discovery REST API request because process is not running or is unresponsive", zap.String("url", url), zap.String("method", method), zap.Error(err))
```

Comment:
`device discovery` or `opentelemetry infinity` ?

Reply:
done, should be opentelemetry